src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -342,31 +336,31 @@ private static ProjectRestoreReference ToProjectRestoreReference(IVsReferenceIte
 
             MSBuildRestoreUtility.ApplyIncludeFlags(
                 dependency,
-                includeAssets: GetPropertyValueOrDefault(item, IncludeAssets),
-                excludeAssets: GetPropertyValueOrDefault(item, ExcludeAssets),
-                privateAssets: GetPropertyValueOrDefault(item, PrivateAssets));
+                includeAssets: GetPropertyValueOrNull(item, IncludeAssets),
+                excludeAssets: GetPropertyValueOrNull(item, ExcludeAssets),
+                privateAssets: GetPropertyValueOrNull(item, PrivateAssets));
 
             return dependency;
         }
 
         private static VersionRange GetVersionRange(IVsReferenceItem item)
         {
-            var versionRange = GetPropertyValueOrDefault(item, "Version");
+            var versionRange = GetPropertyValueOrNull(item, "Version");
 
-            if (!string.IsNullOrEmpty(versionRange))
+            if (versionRange != null)
```

Comment:
MSBuildStringUtility handles string.Empty -> null at a low level (not obvious since it is not part of this PR)